ecosystem/gstreamer_plugin/gst_mtl_common.c

```diff
@@ -391,3 +391,35 @@ gboolean gst_mtl_common_parse_dev_arguments(struct mtl_init_params* mtl_init_par
   gst_mtl_port_idx++;
   return ret;
 }
+
+mtl_handle gst_mtl_common_mtl_init(struct mtl_init_params* p, StDevArgs* devArgs,
+                                   guint* log_level) {
+  struct mtl_init_params mtl_init_params = {0};
+
+  if (!p || !devArgs || !log_level) {
+    GST_ERROR("Invalid input");
+    return NULL;
+  }
+
+  mtl_init_params.num_ports = 0;
+
+  if (gst_mtl_common_parse_dev_arguments(&mtl_init_params, devArgs) == FALSE) {
+    GST_ERROR("Failed to parse dev arguments");
+    return NULL;
+  }
+  mtl_init_params.flags |= MTL_FLAG_BIND_NUMA;
+
+  /*
+   * Log levels range from 1 to LOG_LEVEL_MAX.
+   * We avoid using 0 (DEBUG) in normal scenarios,
+   * so it's acceptable to use 0 as a placeholder.
+   */
+  if (*log_level && *log_level < MTL_LOG_LEVEL_MAX) {
+    mtl_init_params.log_level = *log_level;
+  } else {
+    mtl_init_params.log_level = MTL_LOG_LEVEL_INFO;
```

Comment:
Maybe some log massage that chosen log level is outside supported values? or I think even better decrease the range in `g_param_spec_uint`  so user can chose only supported values